### ISSUE TITLE
cilium agent report "required IPv4 pod CIDR not present in node resource"

### ISSUE THREAD

**Issue by konghui** [cilium/cilium#12956]
## Bug report

**General Information**

- Cilium version (run `cilium version`): 1.8.2
- Kernel version (run `uname -a`): 5.8.0-1.el7.elrepo.x86_64
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...): v1.18.6
- Link to relevant artifacts (policies, deployments scripts, ...)
- Generate and upload a system zip:
```
curl -sLO https://git.io/cilium-sysdump-latest.zip && python cilium-sysdump-latest.zip
```

**How to reproduce the issue**

1. Use such command generate the yaml file:

``` helm template cilium cilium/cilium --version 1.8.2   --namespace kube-system   --set global.hubble.metrics.enabled="{dns,drop,tcp,flow,port-distribution,icmp,http}"  --set global.hubble.enabled=true  --set global.tunnel=disabled  --set global.nodePort.mode=dsr --set global.autoDirectNodeRoutes=true --set global.kubeProxyReplacement=strict --set global.k8sServiceHost=API_SERVER_IP --set global.k8sServicePort=API_SERVER_PORT   --set global.etcd.enabled=true   --set "global.etcd.endpoints[0]=https://etcd-endpoint1:2379"   --set "global.etcd.endpoints[1]=https://etcd-endpoint2:2379" --set global.etcd.ssl=true --set "global.etcd.endpoints[1]=https://etcd-endpoint3:2379" --set global.nodeinit.enabled=true --set global.hostServices.enabled=false --set global.externalIPs.enabled=true --set global.nodePort.enabled=true --set global.hostPort.enabled=true --set global.hubble.listenAddress=":4244" --set global.hubble.relay.enabled=true --set global.hubble.ui.enabled=true  > cilium.yaml
```

2. When I apply this yaml file, the cilium agent report "required IPv4 pod CIDR not present in node resource"
```
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
```

Here is the output of the 
```
kubectl get ciliumnodes.cilium.io 
Name:         l-k8s-06.ops
Namespace:    
Labels:       beta.kubernetes.io/arch=amd64
              beta.kubernetes.io/os=linux
              kubernetes.io/arch=amd64
              kubernetes.io/hostname=l-k8s-06.sg-zl.ops
              kubernetes.io/os=linux
Annotations:  <none>
API Version:  cilium.io/v2
Kind:         CiliumNode
Metadata:
  Creation Timestamp:  2020-08-24T07:14:41Z
  Generation:          2
  Managed Fields:
    API Version:  cilium.io/v2
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:labels:
          .:
          f:beta.kubernetes.io/arch:
          f:beta.kubernetes.io/os:
          f:kubernetes.io/arch:
          f:kubernetes.io/hostname:
          f:kubernetes.io/os:
        f:ownerReferences:
          .:
          k:{"uid":"53f109e2-d8e1-4da1-bc2f-7d25c6d45f86"}:
            .:
            f:apiVersion:
            f:kind:
            f:name:
            f:uid:
      f:spec:
        .:
        f:addresses:
        f:azure:
        f:encryption:
        f:eni:
        f:health:
        f:ipam:
    Manager:      cilium-agent
    Operation:    Update
    Time:         2020-08-24T07:14:41Z
    API Version:  cilium.io/v2
    Fields Type:  FieldsV1
    fieldsV1:
      f:spec:
        f:ipam:
          f:podCIDRs:
      f:status:
        .:
        f:azure:
        f:eni:
        f:ipam:
          .:
          f:operator-status:
            .:
            f:error:
    Manager:    cilium-operator-generic
    Operation:  Update
    Time:       2020-08-24T07:15:11Z
  Owner References:
    API Version:     v1
    Kind:            Node
    Name:            l-k8s-06.ops
    UID:             53f109e2-d8e1-4da1-bc2f-7d25c6d45f86
  Resource Version:  3341241
  Self Link:         /apis/cilium.io/v2/ciliumnodes/l-k8s-06.ops
  UID:               b453bede-6d63-4976-871c-66ca64217858
Spec:
  Addresses:
    Ip:    10.8.192.20
    Type:  InternalIP
  Azure:
  Encryption:
  Eni:
  Health:
  Ipam:
    Pod CID Rs:
      <nil>
Status:
  Azure:
  Eni:
  Ipam:
    Operator - Status:
      Error:  invalid CIDR address: <nil>
Events:       <none>
```

The cilium-operator is no error log output:

```
level=info msg="Skipped reading configuration file" reason="Config File \"cilium-operators\" Not Found in \"[/]\"" subsys=config
level=info msg="  --ccnp-node-status-gc='true'" subsys=cilium-operator-generic
level=info msg="  --cilium-endpoint-gc='true'" subsys=cilium-operator-generic
level=info msg="  --cilium-endpoint-gc-interval='5m0s'" subsys=cilium-operator-generic
level=info msg="  --cluster-id='0'" subsys=cilium-operator-generic
level=info msg="  --cluster-name='kubernetes'" subsys=cilium-operator-generic
level=info msg="  --cluster-pool-ipv4-cidr='10.254.0.0/17'" subsys=cilium-operator-generic
level=info msg="  --cluster-pool-ipv4-mask-size='64'" subsys=cilium-operator-generic
level=info msg="  --cluster-pool-ipv6-cidr=''" subsys=cilium-operator-generic
level=info msg="  --cluster-pool-ipv6-mask-size='112'" subsys=cilium-operator-generic
level=info msg="  --cmdref=''" subsys=cilium-operator-generic
level=info msg="  --cnp-node-status-gc='true'" subsys=cilium-operator-generic
level=info msg="  --cnp-node-status-gc-interval='2m0s'" subsys=cilium-operator-generic
level=info msg="  --cnp-status-update-interval='1s'" subsys=cilium-operator-generic
level=info msg="  --config=''" subsys=cilium-operator-generic
level=info msg="  --config-dir='/tmp/cilium/config-map'" subsys=cilium-operator-generic
level=info msg="  --crd-wait-timeout='5m0s'" subsys=cilium-operator-generic
level=info msg="  --debug='false'" subsys=cilium-operator-generic
level=info msg="  --disable-endpoint-crd='false'" subsys=cilium-operator-generic
level=info msg="  --enable-ipv4='true'" subsys=cilium-operator-generic
level=info msg="  --enable-ipv6='false'" subsys=cilium-operator-generic
level=info msg="  --enable-k8s-api-discovery='false'" subsys=cilium-operator-generic
level=info msg="  --enable-k8s-endpoint-slice='true'" subsys=cilium-operator-generic
level=info msg="  --enable-metrics='false'" subsys=cilium-operator-generic
level=info msg="  --eni-parallel-workers='50'" subsys=cilium-operator-generic
level=info msg="  --identity-allocation-mode='kvstore'" subsys=cilium-operator-generic
level=info msg="  --identity-gc-interval='15m0s'" subsys=cilium-operator-generic
level=info msg="  --identity-gc-rate-interval='1m0s'" subsys=cilium-operator-generic
level=info msg="  --identity-gc-rate-limit='250'" subsys=cilium-operator-generic
level=info msg="  --identity-heartbeat-timeout='30m0s'" subsys=cilium-operator-generic
level=info msg="  --ipam='cluster-pool'" subsys=cilium-operator-generic
level=info msg="  --k8s-api-server=''" subsys=cilium-operator-generic
level=info msg="  --k8s-heartbeat-timeout='30s'" subsys=cilium-operator-generic
level=info msg="  --k8s-kubeconfig-path=''" subsys=cilium-operator-generic
level=info msg="  --k8s-namespace='kube-system'" subsys=cilium-operator-generic
level=info msg="  --kvstore='etcd'" subsys=cilium-operator-generic
level=info msg="  --kvstore-opt='{\"e**How to reproduce the issue**tcd.config\":,\"/var/lib/etcd-config/etcd.config\"}'" subsys=cilium-operator-generic
level=info msg="  --limit-ipam-api-burst='4'" subsys=cilium-operator-generic
level=info msg="  --limit-ipam-api-qps='20'" subsys=cilium-operator-generic
level=info msg="  --log-driver=''" subsys=cilium-operator-generic
level=info msg="  --log-opt='map[]'" subsys=cilium-operator-generic
level=info msg="  --nodes-gc-interval='2m0s'" subsys=cilium-operator-generic
level=info msg="  --operator-api-serve-addr='127.0.0.1:9234'" subsys=cilium-operator-generic
level=info msg="  --operator-prometheus-serve-addr=':6942'" subsys=cilium-operator-generic
level=info msg="  --parallel-alloc-workers='50'" subsys=cilium-operator-generic
level=info msg="  --subnet-ids-filter=''" subsys=cilium-operator-generic
level=info msg="  --subnet-tags-filter='[]'" subsys=cilium-operator-generic
level=info msg="  --synchronize-k8s-nodes='true'" subsys=cilium-operator-generic
level=info msg="  --synchronize-k8s-services='true'" subsys=cilium-operator-generic
level=info msg="  --unmanaged-pod-watcher-interval='15'" subsys=cilium-operator-generic
level=info msg="  --update-ec2-apdater-limit-via-api='false'" subsys=cilium-operator-generic
level=info msg="  --version='false'" subsys=cilium-operator-generic
level=info msg="Cilium Operator 1.8.2 aa42034f0 2020-07-23T15:02:39-07:00 go version go1.14.6 linux/amd64" subsys=cilium-operator-generic
level=info msg="Starting apiserver on address 127.0.0.1:9234" subsys=cilium-operator-generic
level=info msg="Establishing connection to apiserver" host="https://10.8.192.155:6443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s
level=info msg="Starting Operator IP allocator..." subsys=ipam-allocator-operator
level=info msg="Starting to synchronize CiliumNode custom resources..." subsys=cilium-operator-generic
level=info msg="Waiting for CRD ciliumnodes.cilium.io to be available" subsys=cilium-operator-generic
level=info msg="CRD ciliumnodes.cilium.io found" subsys=cilium-operator-generic
level=info msg="Starting to synchronize k8s services to kvstore..." subsys=cilium-operator-generic
level=info msg="Connecting to kvstore..." address= kvstore=etcd subsys=cilium-operator-generic
level=info msg="Connecting to etcd server..." config=/var/lib/etcd-config/etcd.config endpoints="[https://10.8.192.15:2379 https://10.8.192.16:2379 https://10.8.192.17:2379]" subsys=kvstore
level=info msg="Starting to synchronize k8s nodes to kvstore..." subsys=cilium-operator-generic
level=info msg="Restarting unmanaged kube-dns pod kube-system/coredns-784545c97b-d9bzg started 243h44m30.23007378s ago" subsys=cilium-operator-generic
level=info msg="Got lease ID 4b5c73e5d7612822" subsys=kvstore
level=info msg="Got lock lease ID 3fa473e5d75e7815" subsys=kvstore
level=info msg="Initial etcd session established" config=/var/lib/etcd-config/etcd.config endpoints="[https://10.8.192.15:2379 https://10.8.192.16:2379 https://10.8.192.17:2379]" subsys=kvstore
level=info msg="Successfully verified version of etcd endpoint" config=/var/lib/etcd-config/etcd.config endpoints="[https://10.8.192.15:2379 https://10.8.192.16:2379 https://10.8.192.17:2379]" etcdEndpoint="https://10.8.192.15:2379" subsys=kvstore version=3.4.10
level=info msg="Successfully verified version of etcd endpoint" config=/var/lib/etcd-config/etcd.config endpoints="[https://10.8.192.15:2379 https://10.8.192.16:2379 https://10.8.192.17:2379]" etcdEndpoint="https://10.8.192.16:2379" subsys=kvstore version=3.4.10
level=info msg="Successfully verified version of etcd endpoint" config=/var/lib/etcd-config/etcd.config endpoints="[https://10.8.192.15:2379 https://10.8.192.16:2379 https://10.8.192.17:2379]" etcdEndpoint="https://10.8.192.17:2379" subsys=kvstore version=3.4.10
level=info msg="Starting kvstore watchdog with 25s interval..." subsys=cilium-operator-generic
level=info msg="Starting kvstore identity garbage collector with 15m0s interval..." subsys=cilium-operator-generic
level=info msg="Starting to garbage collect stale CiliumEndpoint custom resources..." subsys=cilium-operator-generic
level=info msg="Starting to synchronize Kubernetes services to kvstore" subsys=cilium-operator-generic
level=info msg="Starting to garbage collect stale CiliumNetworkPolicy status field entries..." subsys=cilium-operator-generic
level=info msg="Waiting for CRD ciliumnetworkpolicies.cilium.io to be available" subsys=cilium-operator-generic
level=info msg="CRD ciliumnetworkpolicies.cilium.io found" subsys=cilium-operator-generic
level=info msg="Starting to garbage collect stale CiliumClusterwideNetworkPolicy status field entries..." subsys=cilium-operator-generic
level=info msg="Initialization complete" subsys=cilium-operator-generic
level=info msg="Restarting unmanaged kube-dns pod kube-system/coredns-784545c97b-l7s5z started 243h44m46.672188515s ago" subsys=cilium-operator-generic
level=info msg="Deleted unused allocator master key" id=19646 key=cilium/state/identities/v1/id/19646 subsys=kvstorebackend
level=info msg="Deleted unused allocator master key" id=36878 key=cilium/state/identities/v1/id/36878 subsys=kvstorebackend
```
On the kubernetes I see the podcidr was already allocated

![image](https://user-images.githubusercontent.com/379565/91046513-23749500-e64b-11ea-9733-bdf40e326cfd.png)

**Comment by brb** [comment]
@konghui How did you install Cilium? Can you paste the helm command?

**Comment by bh90210** [comment]
hello @brb 
I have the same problem.

I use `kubeadm init --config kubeadm-config.yaml`

here is my `kubeadm-config.yaml`
```yaml
apiVersion: kubelet.config.k8s.io/v1beta1
kind: KubeletConfiguration
cgroupDriver: docker
---
apiServer:
  timeoutForControlPlane: 4m0s
apiVersion: kubeadm.k8s.io/v1beta2
certificatesDir: /etc/kubernetes/pki
clusterName: dev
controllerManager:
  extraArgs:
    "allocate-node-cidrs": "true"
dns:
  type: CoreDNS
etcd:
  local:
    dataDir: /var/lib/etcd
imageRepository: k8s.gcr.io
kind: ClusterConfiguration
kubernetesVersion: v1.19.0
networking:
  dnsDomain: cluster.local
  podSubnet: 10.100.0.1/24
  serviceSubnet: 10.96.0.0/12
scheduler: {}
---
```

and following cillium's (quick installation)[https://docs.cilium.io/en/stable/gettingstarted/k8s-install-default/] I run 
`kubectl create -f https://raw.githubusercontent.com/cilium/cilium/1.8.2/install/kubernetes/quick-install.yaml`

here's my logs

```
$ kubectl -n kube-system get pods 
NAME                                READY   STATUS              RESTARTS   AGE
cilium-operator-756cc96896-bdx6f    0/1     Pending             0          23h
cilium-rb8qk                        0/1     Running             135        23h
coredns-f9fd979d6-8dgxk             0/1     ContainerCreating   0          23h
coredns-f9fd979d6-z9vv6             0/1     ContainerCreating   0          23h
etcd-desktop-3                      1/1     Running             0          23h
kube-apiserver-desktop-3            1/1     Running             0          23h
kube-controller-manager-desktop-3   1/1     Running             4          23h
kube-proxy-v4tzd                    1/1     Running             0          23h
kube-scheduler-desktop-3            1/1     Running             4          23h
```

```
kubectl -n kube-system logs cilium-rb8qk
level=info msg="Skipped reading configuration file" reason="Config File \"ciliumd\" Not Found in \"[/root]\"" subsys=config
level=info msg="Memory available for map entries (0.003% of 33685528576B): 84213821B" subsys=config
level=info msg="option bpf-ct-global-tcp-max set by dynamic sizing to 295487" subsys=config
level=info msg="option bpf-ct-global-any-max set by dynamic sizing to 147743" subsys=config
level=info msg="option bpf-nat-global-max set by dynamic sizing to 295487" subsys=config
level=info msg="option bpf-neigh-global-max set by dynamic sizing to 295487" subsys=config
level=info msg="option bpf-sock-rev-map-max set by dynamic sizing to 147743" subsys=config
level=info msg="  --agent-health-port='9876'" subsys=daemon
level=info msg="  --agent-labels=''" subsys=daemon
level=info msg="  --allow-icmp-frag-needed='true'" subsys=daemon
level=info msg="  --allow-localhost='auto'" subsys=daemon
level=info msg="  --annotate-k8s-node='true'" subsys=daemon
level=info msg="  --auto-create-cilium-node-resource='true'" subsys=daemon
level=info msg="  --auto-direct-node-routes='false'" subsys=daemon
level=info msg="  --blacklist-conflicting-routes='true'" subsys=daemon
level=info msg="  --bpf-compile-debug='false'" subsys=daemon
level=info msg="  --bpf-ct-global-any-max='262144'" subsys=daemon
level=info msg="  --bpf-ct-global-tcp-max='524288'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-fin='10s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-regular-tcp-syn='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-any='1m0s'" subsys=daemon
level=info msg="  --bpf-ct-timeout-service-tcp='6h0m0s'" subsys=daemon
level=info msg="  --bpf-fragments-map-max='8192'" subsys=daemon
level=info msg="  --bpf-map-dynamic-size-ratio='0.0025'" subsys=daemon
level=info msg="  --bpf-nat-global-max='524288'" subsys=daemon
level=info msg="  --bpf-neigh-global-max='524288'" subsys=daemon
level=info msg="  --bpf-policy-map-max='16384'" subsys=daemon
level=info msg="  --bpf-root=''" subsys=daemon
level=info msg="  --bpf-sock-rev-map-max='262144'" subsys=daemon
level=info msg="  --certificates-directory='/var/run/cilium/certs'" subsys=daemon
level=info msg="  --cgroup-root=''" subsys=daemon
level=info msg="  --cluster-id='0'" subsys=daemon
level=info msg="  --cluster-name='default'" subsys=daemon
level=info msg="  --clustermesh-config='/var/lib/cilium/clustermesh/'" subsys=daemon
level=info msg="  --cmdref=''" subsys=daemon
level=info msg="  --config=''" subsys=daemon
level=info msg="  --config-dir='/tmp/cilium/config-map'" subsys=daemon
level=info msg="  --conntrack-gc-interval='0s'" subsys=daemon
level=info msg="  --datapath-mode='veth'" subsys=daemon
level=info msg="  --debug='false'" subsys=daemon
level=info msg="  --debug-verbose=''" subsys=daemon
level=info msg="  --device=''" subsys=daemon
level=info msg="  --devices=''" subsys=daemon
level=info msg="  --direct-routing-device=''" subsys=daemon
level=info msg="  --disable-cnp-status-updates='true'" subsys=daemon
level=info msg="  --disable-conntrack='false'" subsys=daemon
level=info msg="  --disable-endpoint-crd='false'" subsys=daemon
level=info msg="  --disable-envoy-version-check='false'" subsys=daemon
level=info msg="  --disable-iptables-feeder-rules=''" subsys=daemon
level=info msg="  --disable-ipv4='false'" subsys=daemon
level=info msg="  --disable-k8s-services='false'" subsys=daemon
level=info msg="  --egress-masquerade-interfaces=''" subsys=daemon
level=info msg="  --enable-auto-protect-node-port-range='true'" subsys=daemon
level=info msg="  --enable-bpf-clock-probe='true'" subsys=daemon
level=info msg="  --enable-bpf-masquerade='true'" subsys=daemon
level=info msg="  --enable-endpoint-health-checking='true'" subsys=daemon
level=info msg="  --enable-endpoint-routes='false'" subsys=daemon
level=info msg="  --enable-external-ips='true'" subsys=daemon
level=info msg="  --enable-health-check-nodeport='true'" subsys=daemon
level=info msg="  --enable-health-checking='true'" subsys=daemon
level=info msg="  --enable-host-firewall='false'" subsys=daemon
level=info msg="  --enable-host-port='true'" subsys=daemon
level=info msg="  --enable-host-reachable-services='false'" subsys=daemon
level=info msg="  --enable-hubble='false'" subsys=daemon
level=info msg="  --enable-identity-mark='true'" subsys=daemon
level=info msg="  --enable-ip-masq-agent='false'" subsys=daemon
level=info msg="  --enable-ipsec='false'" subsys=daemon
level=info msg="  --enable-ipv4='true'" subsys=daemon
level=info msg="  --enable-ipv4-fragment-tracking='true'" subsys=daemon
level=info msg="  --enable-ipv6='false'" subsys=daemon
level=info msg="  --enable-k8s-api-discovery='false'" subsys=daemon
level=info msg="  --enable-k8s-endpoint-slice='true'" subsys=daemon
level=info msg="  --enable-k8s-event-handover='false'" subsys=daemon
level=info msg="  --enable-l7-proxy='true'" subsys=daemon
level=info msg="  --enable-local-node-route='true'" subsys=daemon
level=info msg="  --enable-node-port='false'" subsys=daemon
level=info msg="  --enable-policy='default'" subsys=daemon
level=info msg="  --enable-remote-node-identity='true'" subsys=daemon
level=info msg="  --enable-selective-regeneration='true'" subsys=daemon
level=info msg="  --enable-session-affinity='true'" subsys=daemon
level=info msg="  --enable-tracing='false'" subsys=daemon
level=info msg="  --enable-well-known-identities='false'" subsys=daemon
level=info msg="  --enable-xt-socket-fallback='true'" subsys=daemon
level=info msg="  --encrypt-interface=''" subsys=daemon
level=info msg="  --encrypt-node='false'" subsys=daemon
level=info msg="  --endpoint-interface-name-prefix='lxc+'" subsys=daemon
level=info msg="  --endpoint-queue-size='25'" subsys=daemon
level=info msg="  --endpoint-status=''" subsys=daemon
level=info msg="  --envoy-log=''" subsys=daemon
level=info msg="  --exclude-local-address=''" subsys=daemon
level=info msg="  --fixed-identity-mapping='map[]'" subsys=daemon
level=info msg="  --flannel-master-device=''" subsys=daemon
level=info msg="  --flannel-uninstall-on-exit='false'" subsys=daemon
level=info msg="  --force-local-policy-eval-at-source='true'" subsys=daemon
level=info msg="  --host-reachable-services-protos='tcp,udp'" subsys=daemon
level=info msg="  --http-403-msg=''" subsys=daemon
level=info msg="  --http-idle-timeout='0'" subsys=daemon
level=info msg="  --http-max-grpc-timeout='0'" subsys=daemon
level=info msg="  --http-request-timeout='3600'" subsys=daemon
level=info msg="  --http-retry-count='3'" subsys=daemon
level=info msg="  --http-retry-timeout='0'" subsys=daemon
level=info msg="  --hubble-event-queue-size='0'" subsys=daemon
level=info msg="  --hubble-flow-buffer-size='4095'" subsys=daemon
level=info msg="  --hubble-listen-address=''" subsys=daemon
level=info msg="  --hubble-metrics=''" subsys=daemon
level=info msg="  --hubble-metrics-server=''" subsys=daemon
level=info msg="  --hubble-socket-path='/var/run/cilium/hubble.sock'" subsys=daemon
level=info msg="  --identity-allocation-mode='crd'" subsys=daemon
level=info msg="  --identity-change-grace-period='5s'" subsys=daemon
level=info msg="  --install-iptables-rules='true'" subsys=daemon
level=info msg="  --ip-allocation-timeout='2m0s'" subsys=daemon
level=info msg="  --ip-masq-agent-config-path='/etc/config/ip-masq-agent'" subsys=daemon
level=info msg="  --ipam='cluster-pool'" subsys=daemon
level=info msg="  --ipsec-key-file=''" subsys=daemon
level=info msg="  --iptables-lock-timeout='5s'" subsys=daemon
level=info msg="  --ipv4-cluster-cidr-mask-size='8'" subsys=daemon
level=info msg="  --ipv4-node='auto'" subsys=daemon
level=info msg="  --ipv4-pod-subnets=''" subsys=daemon
level=info msg="  --ipv4-range='auto'" subsys=daemon
level=info msg="  --ipv4-service-loopback-address='169.254.42.1'" subsys=daemon
level=info msg="  --ipv4-service-range='auto'" subsys=daemon
level=info msg="  --ipv6-cluster-alloc-cidr='f00d::/64'" subsys=daemon
level=info msg="  --ipv6-node='auto'" subsys=daemon
level=info msg="  --ipv6-pod-subnets=''" subsys=daemon
level=info msg="  --ipv6-range='auto'" subsys=daemon
level=info msg="  --ipv6-service-range='auto'" subsys=daemon
level=info msg="  --ipvlan-master-device='undefined'" subsys=daemon
level=info msg="  --k8s-api-server=''" subsys=daemon
level=info msg="  --k8s-force-json-patch='false'" subsys=daemon
level=info msg="  --k8s-heartbeat-timeout='30s'" subsys=daemon
level=info msg="  --k8s-kubeconfig-path=''" subsys=daemon
level=info msg="  --k8s-namespace='kube-system'" subsys=daemon
level=info msg="  --k8s-require-ipv4-pod-cidr='true'" subsys=daemon
level=info msg="  --k8s-require-ipv6-pod-cidr='false'" subsys=daemon
level=info msg="  --k8s-service-cache-size='128'" subsys=daemon
level=info msg="  --k8s-watcher-endpoint-selector='metadata.name!=kube-scheduler,metadata.name!=kube-controller-manager,metadata.name!=etcd-operator,metadata.name!=gcp-controller-manager'" subsys=daemon
level=info msg="  --k8s-watcher-queue-size='1024'" subsys=daemon
level=info msg="  --keep-bpf-templates='false'" subsys=daemon
level=info msg="  --keep-config='false'" subsys=daemon
level=info msg="  --kube-proxy-replacement='probe'" subsys=daemon
level=info msg="  --kvstore=''" subsys=daemon
level=info msg="  --kvstore-connectivity-timeout='2m0s'" subsys=daemon
level=info msg="  --kvstore-lease-ttl='15m0s'" subsys=daemon
level=info msg="  --kvstore-opt='map[]'" subsys=daemon
level=info msg="  --kvstore-periodic-sync='5m0s'" subsys=daemon
level=info msg="  --label-prefix-file=''" subsys=daemon
level=info msg="  --labels=''" subsys=daemon
level=info msg="  --lib-dir='/var/lib/cilium'" subsys=daemon
level=info msg="  --log-driver=''" subsys=daemon
level=info msg="  --log-opt='map[]'" subsys=daemon
level=info msg="  --log-system-load='false'" subsys=daemon
level=info msg="  --masquerade='true'" subsys=daemon
level=info msg="  --max-controller-interval='0'" subsys=daemon
level=info msg="  --metrics=''" subsys=daemon
level=info msg="  --monitor-aggregation='medium'" subsys=daemon
level=info msg="  --monitor-aggregation-flags='all'" subsys=daemon
level=info msg="  --monitor-aggregation-interval='5s'" subsys=daemon
level=info msg="  --monitor-queue-size='0'" subsys=daemon
level=info msg="  --mtu='0'" subsys=daemon
level=info msg="  --nat46-range='0:0:0:0:0:FFFF::/96'" subsys=daemon
level=info msg="  --native-routing-cidr=''" subsys=daemon
level=info msg="  --node-port-acceleration='disabled'" subsys=daemon
level=info msg="  --node-port-bind-protection='true'" subsys=daemon
level=info msg="  --node-port-mode='snat'" subsys=daemon
level=info msg="  --node-port-range='30000,32767'" subsys=daemon
level=info msg="  --policy-audit-mode='false'" subsys=daemon
level=info msg="  --policy-queue-size='100'" subsys=daemon
level=info msg="  --policy-trigger-interval='1s'" subsys=daemon
level=info msg="  --pprof='false'" subsys=daemon
level=info msg="  --preallocate-bpf-maps='false'" subsys=daemon
level=info msg="  --prefilter-device='undefined'" subsys=daemon
level=info msg="  --prefilter-mode='native'" subsys=daemon
level=info msg="  --prepend-iptables-chains='true'" subsys=daemon
level=info msg="  --prometheus-serve-addr=''" subsys=daemon
level=info msg="  --proxy-connect-timeout='1'" subsys=daemon
level=info msg="  --read-cni-conf=''" subsys=daemon
level=info msg="  --restore='true'" subsys=daemon
level=info msg="  --sidecar-istio-proxy-image='cilium/istio_proxy'" subsys=daemon
level=info msg="  --single-cluster-route='false'" subsys=daemon
level=info msg="  --skip-crd-creation='false'" subsys=daemon
level=info msg="  --socket-path='/var/run/cilium/cilium.sock'" subsys=daemon
level=info msg="  --sockops-enable='false'" subsys=daemon
level=info msg="  --state-dir='/var/run/cilium'" subsys=daemon
level=info msg="  --tofqdns-dns-reject-response-code='refused'" subsys=daemon
level=info msg="  --tofqdns-enable-dns-compression='true'" subsys=daemon
level=info msg="  --tofqdns-enable-poller='false'" subsys=daemon
level=info msg="  --tofqdns-enable-poller-events='true'" subsys=daemon
level=info msg="  --tofqdns-endpoint-max-ip-per-hostname='50'" subsys=daemon
level=info msg="  --tofqdns-max-deferred-connection-deletes='10000'" subsys=daemon
level=info msg="  --tofqdns-min-ttl='0'" subsys=daemon
level=info msg="  --tofqdns-pre-cache=''" subsys=daemon
level=info msg="  --tofqdns-proxy-port='0'" subsys=daemon
level=info msg="  --tofqdns-proxy-response-max-delay='100ms'" subsys=daemon
level=info msg="  --trace-payloadlen='128'" subsys=daemon
level=info msg="  --tunnel='vxlan'" subsys=daemon
level=info msg="  --version='false'" subsys=daemon
level=info msg="  --write-cni-conf-when-ready=''" subsys=daemon
level=info msg="     _ _ _" subsys=daemon
level=info msg=" ___|_| |_|_ _ _____" subsys=daemon
level=info msg="|  _| | | | | |     |" subsys=daemon
level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon
level=info msg="Cilium 1.8.2 aa42034f0 2020-07-23T15:02:39-07:00 go version go1.14.6 linux/amd64" subsys=daemon
level=info msg="cilium-envoy  version: 0a9743dda269a0b0039c9db3cf7e0a637caad7a9/1.13.3/Modified/RELEASE/BoringSSL" subsys=daemon
level=info msg="clang (10.0.0) and kernel (5.4.0) versions: OK!" subsys=linux-datapath
level=info msg="linking environment: OK!" subsys=linux-datapath
level=warning msg="BPF system config check: NOT OK." error="CONFIG_BPF kernel parameter is required" subsys=linux-datapath
level=info msg="Detected mounted BPF filesystem at /sys/fs/bpf" subsys=bpf
level=info msg="Valid label prefix configuration:" subsys=labels-filter
level=info msg=" - :io.kubernetes.pod.namespace" subsys=labels-filter
level=info msg=" - :io.cilium.k8s.namespace.labels" subsys=labels-filter
level=info msg=" - :app.kubernetes.io" subsys=labels-filter
level=info msg=" - !:io.kubernetes" subsys=labels-filter
level=info msg=" - !:kubernetes.io" subsys=labels-filter
level=info msg=" - !:.*beta.kubernetes.io" subsys=labels-filter
level=info msg=" - !:k8s.io" subsys=labels-filter
level=info msg=" - !:pod-template-generation" subsys=labels-filter
level=info msg=" - !:pod-template-hash" subsys=labels-filter
level=info msg=" - !:controller-revision-hash" subsys=labels-filter
level=info msg=" - !:annotation.*" subsys=labels-filter
level=info msg=" - !:etcd_node" subsys=labels-filter
level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.5.0.0/16
level=info msg="Initializing daemon" subsys=daemon
level=info msg="Establishing connection to apiserver" host="https://10.96.0.1:443" subsys=k8s
level=info msg="Connected to apiserver" subsys=k8s
level=info msg="Inheriting MTU from external network interface" device=enp4s0 ipAddr=192.168.1.5 mtu=1500 subsys=mtu
level=info msg="Trying to auto-enable \"enable-node-port\", \"enable-external-ips\", \"enable-host-reachable-services\", \"enable-host-port\", \"enable-session-affinity\" features" subsys=daemon
level=warning msg="Session affinity for host reachable services needs kernel 5.7.0 or newer to work properly when accessed from inside cluster: the same service endpoint will be selected from all network namespaces on the host." subsys=daemon
level=info msg="Restored services from maps" failed=0 restored=0 subsys=service
level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=CiliumNetworkPolicy/v2 subsys=k8s
level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=CiliumClusterwideNetworkPolicy/v2 subsys=k8s
level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=v2.CiliumEndpoint subsys=k8s
level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=v2.CiliumNode subsys=k8s
level=info msg="CRD (CustomResourceDefinition) is installed and up-to-date" name=v2.CiliumIdentity subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
level=info msg="Retrieved node information from cilium node" nodeName=desktop-3 subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
```

**Comment by brb** [comment]
@bh90210 Your `cilium-operator` is in the pending state. Check why it couldn't run. Fixing this should resolve your issue.

**Comment by bh90210** [comment]
uh, initially I assumed the operator will start after whatever process was going on in the `cilium-rb8qk`.
But yes, obviously makes more sense that the operator has to be online first. 

Turns out I could not get any logs since the pod was not running yet. I used the `kubectl describe pod`.
A google search later (I am sorry I have lost the `describe log`) my problem was solved by running `kubectl taint nodes --all node-role.kubernetes.io/master-` as [described here](https://github.com/calebhailey/homelab/issues/3#issuecomment-569543391)

So the above described problem is fixed for me. Thanks for the quick reply, sorry for the not so quick feedback.


Now I have a problem with CoreDNS (I share some logs beneath) but probably is unrelated to cilium.

```
$ kubectl -n kube-system get pods 
NAME                                READY   STATUS    RESTARTS   AGE
cilium-operator-756cc96896-6rv4h    1/1     Running   0          5h27m
cilium-pvchk                        1/1     Running   1          5h27m
coredns-f9fd979d6-tgvfj             0/1     Running   0          5h27m
coredns-f9fd979d6-xckks             0/1     Running   0          5h27m
etcd-desktop-3                      1/1     Running   0          5h28m
kube-apiserver-desktop-3            1/1     Running   0          5h28m
kube-controller-manager-desktop-3   1/1     Running   1          5h28m
kube-proxy-67km5                    1/1     Running   0          5h27m
kube-scheduler-desktop-3            1/1     Running   1          5h28m
```

```
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
I0902 14:41:37.485366       1 trace.go:116] Trace[1616138287]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125 (started: 2020-09-02 14:41:07.484958035 +0000 UTC m=+136.124296345) (total time: 30.000383878s):
Trace[1616138287]: [30.000383878s] [30.000383878s] END
E0902 14:41:37.485381       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Service: Get "https://10.96.0.1:443/api/v1/services?limit=500&resourceVersion=0": dial tcp 10.96.0.1:443: i/o timeout
I0902 14:41:37.800864       1 trace.go:116] Trace[1858292790]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125 (started: 2020-09-02 14:41:07.80057439 +0000 UTC m=+136.439912689) (total time: 30.000267008s):
Trace[1858292790]: [30.000267008s] [30.000267008s] END
E0902 14:41:37.800877       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Namespace: Get "https://10.96.0.1:443/api/v1/namespaces?limit=500&resourceVersion=0": dial tcp 10.96.0.1:443: i/o timeout
I0902 14:41:41.072049       1 trace.go:116] Trace[1124895541]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125 (started: 2020-09-02 14:41:11.071687068 +0000 UTC m=+139.711025377) (total time: 30.000340975s):
Trace[1124895541]: [30.000340975s] [30.000340975s] END
E0902 14:41:41.072060       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.18.3/tools/cache/reflector.go:125: Failed to list *v1.Endpoints: Get "https://10.96.0.1:443/api/v1/endpoints?limit=500&resourceVersion=0": dial tcp 10.96.0.1:443: i/o timeout
[INFO] plugin/ready: Still waiting on: "kubernetes"
[INFO] plugin/ready: Still waiting on: "kubernetes"
```

**Comment by IgorSor** [comment]
Hello @brb, we have the same problem in a different case.
We are living in eks, using helm and trying to migrate from release 1.7.3 to 1.8.2 and it seems like everything fine until we need to recreate a node in aws.
And after that we have a cilium agent with the same error:
`level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s`
logfile from the agent in attachment
[agent.log](https://github.com/cilium/cilium/files/5183743/agent.log)

**Comment by aanm** [comment]
@IgorSor do you have cilium-operator running? Can you describe all steps that you have performed so we can replicate the issue on our side?

**Comment by aanm** [comment]
@konghui are you still facing issues after making sure the cilium-operator is running?

**Comment by IgorSor** [comment]
> @IgorSor do you have cilium-operator running?
Yes, we have.
> Can you describe all steps that you have performed so we can replicate the issue on our side?
We are installing cilium 1.8.2 on eks cluster in aws. After that, we are terminating one of the nodes.
After that, new node will bring back to life. On that node we will have this problem with cilium agent.
If it needed, I can attach logs from cilium-operator.

**Comment by qmonnet** [comment]
Maybe related to #13193.

**Comment by konghui** [comment]
It's my fault I misconfig the nodemasksize grate than 32 for ipv4. It cause ipam return nil and cause this problem.
